.github/workflows/README.md

```diff
@@ -0,0 +1,37 @@
+# Envoy Github workflows
+
+## Trusted workflows
+
+Github workflows that are **not** triggered by a `pull_request` generally run with
+the repository context/permissions.
+
+In various ways, these workflows can be triggered as the result of a `pull_request`
+and/or be made to run untrusted code (ie PR code).
+
+This can be useful, but carries significant risks.
+
+In particular this can effect:
+
+- `pull_request_target`
+- `workflow_run`
+- `workflow_dispatch`
+
+Do not use these trigger events unless they are required.
+
+## Restrict access to trusted workflows
+
+It is important to restrict who can trigger these types of workflow.
+
+Do not allow any bots or app users to do so, unless this is specifically required.
+
+## Trusted/untrusted CI jobs
+
+If a trusted workflow is used to run untrusted code, then the entire job that runs this code
+should be treated as untrusted.
+
+In this case, it is **essential** to ensure:
+
+- restrictive global read permissions
+- no global write permissions
```

Comment:
probably these should be required everywhere

Reply:
Agreed!